### PR TITLE
docs: add summary to Microsoft.Extensions.Options

### PR DESCRIPTION
**Pull request by ikesnowy**
## Summary

Just add the namespace document as the issue says it should be committed directly to this repo.

Contributes to https://github.com/dotnet/runtime/issues/43919

### CODE REVIEW

**Comment by MSDN-WhiteKnight** [comment]
Consider also adding remarks with a link to related documentation (such as <https://docs.microsoft.com/dotnet/core/extensions/options>)

**Comment by ikesnowy** [comment]
> Consider also adding remarks with a link to related documentation (such as https://docs.microsoft.com/dotnet/core/extensions/options)

Added, thanks!

File: xml/ns-Microsoft.Extensions.Options.xml [hunk 1]
```diff
@@ -1,6 +1,13 @@
 <Namespace Name="Microsoft.Extensions.Options">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Provides types used for implementing options pattern.</summary>
```
Comment (gewarren) [inline xml/ns-Microsoft.Extensions.Options.xml:3]:
```suggestion
    <summary>Provides types used for implementing the options pattern.</summary>
```

File: xml/ns-Microsoft.Extensions.Options.xml [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+    <remarks>
+      <format type="text/markdown"><![CDATA[
+
+## Remarks
+ For more information about implementing options pattern, see the [Options Pattern](/dotnet/core/extensions/options) documentation.
```
Comment (gewarren) [inline xml/ns-Microsoft.Extensions.Options.xml:8]:
```suggestion

For more information about implementing the options pattern, see [Options pattern in .NET](/dotnet/core/extensions/options).
```

**Review by gewarren — COMMENTED** [review]
Thanks for adding this!

**Review by tarekgh — APPROVED** [review]

**Pull request merged**